onto-viewer-config-loader/src/main/java/org/edmcouncil/spec/ontoviewer/configloader/configuration/service/YamlFileBasedConfigurationService.java

```diff
@@ -161,24 +166,65 @@ private String readConfigForFileName(String configFileName) {
     return "";
   }
 
-  private String readConfigContent(Entry<String, String> configEntry) {
+  private String readRemoteConfigContent(Entry<String, String> configEntry) {
+    var configContent = downloadYamlFileContent(configEntry.getValue());
+
     try {
-      var configContent = IOUtils.toString(new URL(configEntry.getValue()), StandardCharsets.UTF_8);
-      overrideConfigContent(configEntry.getKey(), configContent);
-    } catch (IOException ex) {
-      LOGGER.warn("Exception thrown while loading configuration from URL '{}'. Details: {}",
-          configEntry.getValue(), ex.getMessage(), ex);
+      // We want to check if the input config content is a valid YAML
+      var yaml = new Yaml();
+      yaml.load(configContent);
+    } catch (YAMLException ex) {
+      LOGGER.warn("YAML config file '{}' from URL '{}' isn't correct. Ignoring it. YAML reading exception: {}",
+          configEntry.getKey(),
+          configEntry.getValue(),
+          ex.getMessage());
+      configContent = "";
+    }
+
+    overrideConfigContent(configEntry.getKey(), configContent);
+    return configContent;
+  }
+
+  private String downloadYamlFileContent(String url) {
+    Request request = new Request.Builder()
+        .url(url)
+        .get()
+        .addHeader(ACCEPT_HEADER, YAML_MIME_TYPE)
+        .build();
+
+    try (Response response = httpClient.newCall(request).execute()) {
+      var responseCode = response.code();
+      var responseBody = response.body().string();
+      if (!response.isSuccessful()) {
+        LOGGER.warn("Request downloading configuration file from URL '{}' wasn't successful. "
```

Comment:
If the file does not exist, the WARN log appears, but additionaly the WARN log appears:

> 2023-02-17 16:30:14,766 WARN  [UpdateThread-0] org.edmcouncil.spec.ontoviewer.configloader.configuration.service.YamlFileBasedConfigurationService: Exception thrown while reading config content from path 'config/integration_config.yaml'. Details: config/integration_config.yaml
> java.nio.file.NoSuchFileException: config/integration_config.yaml

generated by:
https://github.com/edmcouncil/onto-viewer/blob/fb7d453bb33aead7cbfec72f4f200125dc8fe0d2/onto-viewer-config-loader/src/main/java/org/edmcouncil/spec/ontoviewer/configloader/configuration/service/YamlFileBasedConfigurationService.java#L163-L164

This situation should be handled differently

Reply:
As for:

> the WARN log appears, but additionaly the WARN log appears

@przemekgradzki this sounds cryptic - could you elaborate?

Reply:
As for:

>This situation should be handled differently.

@przemekgradzki could you suggest a solution, i.e., how exactly to treat this differently?

Reply:
comment updated

Reply:
We agreed to live with this inconvenience for now.